satpy/multiscene.py

```diff
@@ -69,29 +69,72 @@ def timeseries(datasets):
     return res
 
 
-def add_group_aliases(scenes, groups):
-    """Add aliases for the groups datasets belong to."""
+def group_datasets_in_scenes(scenes, groups):
+    """Group different datasets in multiple scenes by adding aliases.
+
+    Args:
+        scenes (iterable): Scenes to be processed.
+        groups (dict): Groups of datasets that shall be treated equally by
+            MultiScene. Keys specify the groups, values specify the dataset
+            names to be grouped. For example::
+
+            from satpy import DataQuery
+            groups = {DataQuery(name='odd'): ['ds1', 'ds3'],
+                      DataQuery(name='even'): ['ds2', 'ds4']}
+    """
     for scene in scenes:
-        scene = scene.copy()
-        for group_id, member_names in groups.items():
-            # Find out whether one of the datasets in this scene belongs
-            # to this group
-            member_ids = [scene[name].attrs['_satpy_id']
-                          for name in member_names if name in scene]
-
-            # Add an alias for the group it belongs to
-            if len(member_ids) == 1:
-                member_id = member_ids[0]
-                new_ds = scene[member_id].copy()
-                new_ds.attrs.update(group_id.to_dict())
-                scene[group_id] = new_ds
-            elif len(member_ids) > 1:
-                raise ValueError('Cannot add multiple datasets from the same '
-                                 'scene to a group')
-            else:
-                # Datasets in this scene don't belong to any group
-                pass
-        yield scene
+        grp = GroupAliasGenerator(scene, groups)
+        yield grp.duplicate_datasets_with_group_alias()
+
+
+class GroupAliasGenerator:
+    """Add group aliases to a scene."""
+
+    def __init__(self, scene, groups):
+        """Initialize the alias generator."""
+        self.scene = scene.copy()
+        self.groups = groups
+
+    def duplicate_datasets_with_group_alias(self):
+        """Duplicate datasets to be grouped with a group alias."""
+        for group_id, group_members in self.groups.items():
+            self._duplicate_dataset_with_group_alias(group_id, group_members)
+        return self.scene
+
+    def _duplicate_dataset_with_group_alias(self, group_id, group_members):
+        member_ids = self._get_dataset_id_of_group_members_in_scene(group_members)
+        if len(member_ids) == 1:
+            self._duplicate_dataset_with_different_id(
+                dataset_id=member_ids[0],
+                alias_id=group_id,
+            )
+        elif len(member_ids) > 1:
+            raise ValueError('Cannot add multiple datasets from a scene '
+                             'to the same group')
```

Comment:
Is there a case where `member_ids` is 0 or negative? If not, maybe the check for `> 1` could go first in this method (raising the exception) and the `== 1` case would be all that's left (less indentation).